lib/carbon/database.py

```diff
@@ -54,6 +59,26 @@ def validateArchiveList(self, archiveList):
     "Validate that the database can handle the given archiveList."
     pass
 
+  def tag(self, metric, httpRequest=None):
+    log.msg("Tagging %s" % metric)
+    t = time.time()
+
+    if not httpRequest:
+      log.msg("Error tagging %s: %s" % (metric, 'missing httpRequest param'))
+      return None
+
+    def successHandler(result, *args, **kw):
+      log.msg("Tagged %s: %s in %s" % (metric, result, time.time() - t))
+      return result
+
+    def errorHandler(err):
+      log.msg("Error tagging %s: %s" % (metric, err))
+
+    return httpRequest(
```

Comment:
Maybe don't return anything here ?

Reply:
Is there a reason not to return it? Right now writer.py doesn't do anything with it but in future maybe it would be useful

Reply:
Because it means that the API needs to return an HTTP request if you do that. Which clearly won't be the case in all implementations